File: SXmatrices.m
% Sampling rate
sr = 22050;

% Load audio file
filepath = './phase-recon-mel/billy.wav';
[y, fs] = audioread(filepath);

% Resample if necessary
if fs ~= sr
    y = resample(y, sr, fs);
end

% Trim to 20 seconds
y = y(1:min(end, 20 * sr));

% Compute STFT
n_fft = 1024; % FFT window size
hop_length = 256; % Hop size
window = hann(n_fft); % Hann window

% Compute STFT using the spectrogram function
X = stft(y, sr, 'Window', window, 'OverlapLength', hop_length, 'FFTLength', n_fft);

% Compute magnitude and phase
S = abs(X); % Magnitude spectrogram
phase = angle(X); % Phase spectrogram

% Print matrices
disp('Magnitude Spectrogram (S):');
disp(S);

disp('Phase Spectrogram:');
disp(phase);

File: phase-recon-mel/SXmatrices.py
import numpy as np
import librosa

sr=22050

# Load audio file
# Load audio file
filepath = './billy.wav'
y, fs = librosa.load(filepath, sr=sr, duration=20.0)

# Compute STFT
n_fft = 1024  # FFT window size
hop_length = 256  # Hop size
X = librosa.stft(y, n_fft=n_fft, hop_length=hop_length)  # Complex STFT matrix

# Compute magnitude and phase
S = np.abs(X)  # Magnitude spectrogram
phase = np.angle(X)  # Phase spectrogram

# Print matrices
np.set_printoptions(suppress=True, precision=4)  # Format for readability
print("Magnitude Spectrogram (S):")
print(S)

print("\nPhase Spectrogram:")
print(phase)
